lib/upgradedb/postgres.2022062808.php

```diff
@@ -23,11 +23,15 @@
 
 $this->BeginTrans();
 
-if (!$this->ResourceExists('voip_cdr_type_uniqueid_ukey', LMSDB::RESOURCE_TYPE_CONSTRAINT)) {
+if ($this->ResourceExists(
+    'voip_cdr_uniqueid_key',
+    LMSDB::RESOURCE_TYPE_CONSTRAINT
+)) {
```

Comment:
Ja użyłem wcześniej z jakiegoś powodu sprawdzania czy istnieje `voip_cdr_type_unqiueid_ukey` (nie przypadkowo), a Ty stwierdziłeś, że u Ciebie to  jest pod inną nazwą i dlatego ma być tak jak u Ciebie. Co w sytuacji, gdy na niektórych instalacjach na tym etapie będzie istniał `voip_cdr_type_uniqueid_ukey', a na innych `voip_cdr_uniqueid_key`?

Reply:
Użyłeś sprawdzania jednego klucza (voip_cdr_type_uniqueid_ukey) a dropowałeś inny (voip_cdr_uniqueid_key) to z jakiegoś powodu zamierzone?

Reply:
`voip_cdr_type_unqiueid_ukey` również nie ma nigdzie choć to wydaje się literówka bo nie ma nazwy tej w kodzie

Reply:
> Co w sytuacji, gdy na niektórych instalacjach na tym etapie będzie istniał voip_cdr_type_uniqueid_ukey, a na innych voip_cdr_uniqueid_key?

Ale jak może on istnieć na tym etapie skoro nigdzie go nie ma?

Reply:
> Ale jak może on istnieć na tym etapie skoro nigdzie go nie ma?

Kiedyś nie nadawaliśmy jawnych nazw constraintom i w efekcie silnik DB im nadawał własne, domyślne nazwy. Konstrukcja domyślnej nazwy potrafiła się zmienić między wersjami silników DB.

Reply:
> Lepiej byłoby użyć ResourceExists('customerassignments_customergroupid_key', ...), żeby wyznaczyć pod jaką nazwą figuruje ten constraint (bo może tak być, że jednak jest, ale pod inną nazwą).

`ResourceExists()` nie zwraca nazw constraintów a jedynie ich liczbę. Więc nie da się wyszukać nazwy za pomocą `ResourceExists()` bez jej rozbudowy. Chyba, że się mylę?

Reply:
> > Lepiej byłoby użyć ResourceExists('customerassignments_customergroupid_key', ...), żeby wyznaczyć pod jaką nazwą figuruje ten constraint (bo może tak być, że jednak jest, ale pod inną nazwą).
> 
> `ResourceExists()` nie zwraca nazw constraintów a jedynie ich liczbę. Więc nie da się wyszukać nazwy za pomocą `ResourceExists()` bez jej rozbudowy. Chyba, że się mylę?

A po co? Nie bardzo rozumiem o jakiej nazwie piszesz? Chodzi przecież o "spróbkowanie" paru potencjalnych nazw i jeśli jakaś się znajdzie to jej warunkowe usunięcie.

> Ja nie mam danych nt historycznych nazw kluczy nadawanych automatycznie w różnych implementacjach MySQL/MariaDB - wiec nie domyślę się.

I nie musisz - wystarczy, że założysz, że skoro ja użyłem innej nazwy niż Ty chcesz teraz użyć, tzn. że zdarzyło mi się napotkać CONSTRAINTA pod inną nazwą niż Ty napotkałeś, więc skoro obydwaj napotkaliśmy pod dwoma różnymi nazwami to pomyśl to z tego wynika...

Reply:
@chilek Tak jest ok?

Reply:
@chilek Tak jest już ok?

Reply:
@chilek Tak jest już ok?